lib/ProxyFactory.php

```diff
@@ -95,6 +91,9 @@ public function createProxyForNode($fromDocument, NodeInterface $targetNode)
             return $document;
         }
 
+        $eventDispatcher = $this->dispatcher;
+        $registry = $this->registry;
```

Comment:
You couldn't do that in earlier versions of PHP, and I am not sure about the behavior now.